src/Stripe/Infrastructure/Requestor.cs

```diff
@@ -1,48 +1,74 @@
+using System.Reflection;
+using System.Threading.Tasks;
 
 namespace Stripe
 {
     internal static class Requestor
     {
+        
+        /// <summary>
+        /// The internal HttpClient instance to be used by the Requestor.
+        /// </summary>
+        internal static HttpClient HttpClient { get; }
+		
+		internal static string Version { get; }
+
+        /// <summary>
+        /// Initializes the HTTP request internals.
+        /// </summary>
+        static Requestor()
+        {
+            var handler = new HttpClientHandler();
+            if (handler.SupportsAutomaticDecompression)
+            {
+                handler.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
+            }
+
+            HttpClient = new HttpClient(handler);
+			
+            var thisAssembly = typeof(Requestor).GetTypeInfo().Assembly;
+            var thisAssemblyName = new AssemblyName(thisAssembly.FullName);
+            Version = thisAssemblyName.Version.ToString(3);
+        }
+
         public static string GetString(string url, StripeRequestOptions requestOptions)
         {
-            var wr = GetWebRequest(url, "GET", requestOptions);
+            var wr = GetRequestMessage(url, HttpMethod.Get, requestOptions);
 
-            return ExecuteWebRequest(wr);
+            return ExecuteRequest(wr).Result;
         }
 
         public static string PostString(string url, StripeRequestOptions requestOptions)
         {
-            var wr = GetWebRequest(url, "POST", requestOptions);
+            var wr = GetRequestMessage(url, HttpMethod.Post, requestOptions);
 
-            return ExecuteWebRequest(wr);
+            return ExecuteRequest(wr).Result;
         }
 
         public static string Delete(string url, StripeRequestOptions requestOptions)
         {
-            var wr = GetWebRequest(url, "DELETE", requestOptions);
+            var wr = GetRequestMessage(url, HttpMethod.Delete, requestOptions);
 
-            return ExecuteWebRequest(wr);
+            return ExecuteRequest(wr).Result;
         }
 
         public static string PostStringBearer(string url, StripeRequestOptions requestOptions)
         {
-            var wr = GetWebRequest(url, "POST", requestOptions, true);
+            var wr = GetRequestMessage(url, HttpMethod.Post, requestOptions, true);
 
-            return ExecuteWebRequest(wr);
+            return ExecuteRequest(wr).Result;
         }
 
-        internal static WebRequest GetWebRequest(string url, string method, StripeRequestOptions requestOptions, bool useBearer = false)
+        internal static HttpRequestMessage GetRequestMessage(string url, HttpMethod method, StripeRequestOptions requestOptions, bool useBearer = false)
         {
             requestOptions.ApiKey = requestOptions.ApiKey ?? StripeConfiguration.GetApiKey();
 
-            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
```

Comment:
That functionality is not available (AFAIK) on System.Net.HttpClient. We can search for a suitable replacement... there may be a HttpClientHandler that supports it.